addons/point_of_sale/static/src/js/models.js

```diff
@@ -2131,7 +2107,8 @@ exports.Order = Backbone.Model.extend({
             pos_session_id: this.pos_session_id,
             pricelist_id: this.pricelist ? this.pricelist.id : false,
             partner_id: this.get_client() ? this.get_client().id : false,
-            user_id: this.pos.get_cashier().id,
+            user_id: this.pos.user.id,
```

Comment:
wrong it should be employee.user_id if employee is selected and has a user_id else this.pos.user.id